include/bluetooth/mesh/sensor.h

```diff
@@ -24,7 +24,9 @@ extern "C" {
 
 /** Largest period divisor value allowed. */
 #define BT_MESH_SENSOR_PERIOD_DIV_MAX 15
-/** Largest sensor interval allowed (in seconds). */
+/** Largest sensor interval allowed. The value is represented as
+ *  2 to the power of N milliseconds.
```

Comment:
Although absolutely grammatically correct, I wonder if we should consider replacing with 2^N miliseconds. Short and clear. I'll leave it up to you, just a suggestion.
And the same for line 383.